python/pyspark/sql/tests/test_pandas_udf_scalar.py

```diff
@@ -897,6 +897,30 @@ def test_timestamp_dst(self):
             result = df.withColumn('time', foo_udf(df.time))
             self.assertEquals(df.collect(), result.collect())
 
+    def test_createDateFrame_with_category_type(self):
```

Comment:
This test module is for `pandas_udf`s, not for `createDataFrame`. We do need to add a `pandas_udf` that tests this. The user would specify a return type of `string` and then return a categorical pandas.Series that has string categories. For example:

```python
@pandas_udf('string')
def f(x):
    return x.astype('category')

pdf = pd.DataFrame({"A": [u"a", u"b", u"c", u"a"]})
df = spark.createDataFrame(pdf).withColumn("B", f(col("A")))
result = df.toPandas()
# Check result "B" is equal to "A"
```

Reply:
aha, got it. fixed